test/cfa/ntp_conf_test.rb

```diff
@@ -0,0 +1,305 @@
+    tree.add("iburst", nil)
+    tree
+  end
+
+  let(:augeas_tree_value) do
+    CFA::AugeasTreeValue.new(augeas_options, "4.pool.ntp.org")
+  end
+
+  let(:augeas_element) do
+    {
+      key:   "server[]",
+      value: augeas_tree_value
+    }
+  end
+
+  subject(:record) { described_class.new_from_augeas(augeas_element) }
+
+  describe ".new_from_augeas" do
+    it "creates a record of the correct class" do
+      expect(record).to be_a(CFA::NtpConf::ServerRecord)
+    end
+
+    it "creates the record with correct data" do
+      expect(record.augeas).to eq(augeas_element)
+    end
+  end
+
+  describe "#value" do
+    it "obtains the value of the record" do
+      expect(record.value).to eq(augeas_element[:value].value)
+    end
+  end
+
+  describe "#value=" do
+    it "sets the value of the record" do
+      value = "1.pool.ntp.org"
+      record.value = value
+      expect(record.value).to eq(value)
+    end
+  end
+
+  describe "#comment" do
+    context "when the record has not comment" do
+      it "obtains nil" do
+        expect(record.comment).to eq(nil)
+      end
+    end
+
+    context "when the record has comment" do
+      let(:augeas_options) do
+        tree = CFA::AugeasTree.new
+        tree.add("#comment", "sample comment")
+        tree
+      end
+
+      it "obtains the comment" do
+        expect(record.comment).to eq("sample comment")
+      end
+    end
+  end
+
+  describe "#comment=" do
+    it "sets a comment to the record" do
+      comment = "sample comment"
+      record.comment = comment
+      expect(record.comment).to eq(comment)
+    end
+  end
+
+  describe "#==" do
+    it "returns true for equal records" do
+      equal_record = described_class.new_from_augeas(augeas_element)
+      expect(record == equal_record).to be(true)
+    end
+
+    it "returns false for different records" do
+      different_record = CFA::NtpConf::Record.new(key: "server[]")
+      different_record.value = "10.10.10.10"
+      expect(record == different_record).to be(false)
+    end
+  end
+
+  describe "#type" do
+    it "obtains the type of the record" do
+      expect(record.type).to eq("server")
+    end
+  end
+
+  describe "#raw_options" do
+    it "obtains options as string" do
+      expect(record.raw_options).to eq("iburst")
+    end
+  end
+
+  describe "#raw_options=" do
+    it "sets options from a string" do
+      record.raw_options = "iburst prefer"
+      expect(record.options).to eq(["iburst", "prefer"])
+    end
+  end
+end
+
+describe CFA::NtpConf::CommandRecord do
+  let(:augeas_element) do
+    tree = CFA::AugeasTree.new
+    tree.add("iburst", nil)
+    value = CFA::AugeasTreeValue.new(tree, "4.pool.ntp.org")
+    { key: "server[]", value: value }
+  end
+
+  subject(:record) { described_class.new_from_augeas(augeas_element) }
+
+  describe "#options" do
+    it "obtains the options of the record" do
+      expect(record.options).to eq(["iburst"])
+    end
+  end
+
+  describe "#options=" do
```

Comment:
and vice verse for setting options. I would like to see that new options are saved and then verify that it looks as expected in ntp.conf. ( actually that is purpose of memoryFile file handler, for easier testing read/write )